gcp/gcp.go

```diff
@@ -1,27 +1,25 @@
 package gcp
 
 import (
-	"cloud.google.com/go/pubsub"
-	"google.golang.org/api/option"
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/cloudchacho/hedwig-go"
 )
 
 // This should satify interface for FirehoseBackend
 type Backend struct {
-	client   *pubsub.Client
-	settings Settings
-}
-
-// SubscriptionProject represents a tuple of subscription name and project for cross-project Google subscriptions
-type SubscriptionProject struct {
-	// Subscription name
-	Subscription string
-
-	// ProjectID
-	ProjectID string
+	GcsClient        *storage.Client
+	FirehoseSettings FirehoseSettings
 }
 
 // Settings for Hedwig firehose
-type Settings struct {
+type FirehoseSettings struct {
```

Comment:
kinda not liking the `Firehose` prefix in naming. Since everything is under `firehose` app, perhaps we can omit the prefix.